external/llvm-project/mlir/lib/Dialect/XModel/Pipelines/Pipelines.cpp

```diff
@@ -15,35 +15,72 @@
 // limitations under the License.
 // =============================================================================
 //
-// This interface adds the Rock compilation pipeline for various flows but
+// This interface adds the XModel compilation pipeline for various flows but
 // keeping a unified ordering of the pipeline.
 //
 //===----------------------------------------------------------------------===//
 
-#include "mlir/Dialect/Rock/Pipelines/XMIRPipelines.h"
+#include "mlir/Dialect/XModel/Pipelines/Pipelines.h"
 
-#include "mlir/Dialect/Func/IR/FuncOps.h"
-#include "mlir/Dialect/Rock/Passes.h"
+#include "mlir/Dialect/Bufferization/Transforms/OneShotAnalysis.h"
+#include "mlir/Dialect/XModel/Transforms/Passes.h"
 #include "mlir/InitAllDialects.h"
 #include "mlir/InitAllPasses.h"
 #include "mlir/Pass/PassManager.h"
 #include "mlir/Pass/PassRegistry.h"
 #include "mlir/Transforms/Passes.h"
 
+#include "llvm/Support/TargetSelect.h"
+
 using namespace mlir;
 
+//===- Consolidate the XModel Pipelines here ---------------------===//
+
+void xmodel::buildGraphPipeline(OpPassManager &pm,
+                                const xmodel::GraphOptions &options) {
+  // TOSA partitioning pass
+  // make 'kernel' funcs with tosa dataflow
+  /* rocmlir-opt --tosa-layerwise-constant-fold --tosa-make-broadcastable
+         --tosa-optional-decompositions --tosa-partition
+   */
+  pm.addNestedPass<func::FuncOp>(tosa::createTosaLayerwiseConstantFoldPass());
+  pm.addNestedPass<func::FuncOp>(tosa::createTosaMakeBroadcastablePass());
+  pm.addNestedPass<func::FuncOp>(tosa::createTosaOptionalDecompositions());
+  pm.addPass(
+      tosa::createTosaPartition({{"tosa.conv2d", "tosa.depthwise_conv2d"}}));
+
+  // make async kernel launch's
+  /* rocmlir-opt --xmodel-async-graph
+   */
+  pm.addNestedPass<func::FuncOp>(createXModelAsyncGraphPass());
+
+  // clone 'kernel' funcs into __rock module
+  /* rocmlir-opt --xmodel-target-kernels
```

Comment:
We can probably call this `mlir-opt` given that it's in external now? Also, does it still go to `__rock`?